### PR TITLE
Add Progress Bar component

### PR DESCRIPTION
**Pull request by hcourthias**
## Checklist before requesting a review
- [x] Working on iOS
- [x] Working on Android
- [x] Integration tests added
- [x] Visual regressions screenshots are up to date
- [ ] Design validated

### CODE REVIEW

File: packages/streamline/src/components/progress-bar/progress-bar.utils.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    case 'primary':
+      return { backgroundColor: 'GREY_100', backgroundOpacity: 1 };
+    case 'light':
+      return { backgroundColor: `PURE_WHITE_1000`, backgroundOpacity: 0.1 };
+    case 'dark':
+      return { backgroundColor: `GREY_1000`, backgroundOpacity: 0.1 };
+    default:
+      return { backgroundColor: `GREY_100`, backgroundOpacity: 1 };
+  }
+};
```
Comment (judastheo) [inline packages/streamline/src/components/progress-bar/progress-bar.utils.ts:33]: @kimak convinced to not put test for trivial functions such as the theme switch. @hcourthias WDYT?

**Review by judastheo — APPROVED** [review]

File: packages/streamline/src/components/progress-bar/progress-bar.utils.ts (hunk 1 above)
Comment (judastheo) [reply packages/streamline/src/components/progress-bar/progress-bar.utils.ts:33]: https://github.com/GetLuko/streamline/pull/27#discussion_r1151723421

Comment (hcourthias) [reply packages/streamline/src/components/progress-bar/progress-bar.utils.ts:33]: I find them useful and this is very fast to write

Comment (hcourthias) [reply packages/streamline/src/components/progress-bar/progress-bar.utils.ts:33]: to match the rest of the project, I will remove them

**Pull request merged**